src/components/ExchangePage/index.tsx

```diff
@@ -468,7 +460,7 @@ function ExchangePage({ sendingInput = false, history, params }: ExchangePagePro
           sending ? recipient : account,
           deadlineFromNow
         ]
-        value = hex(slippageAdjustedAmounts[Field.INPUT].raw)
+        value = BigNumber.from(slippageAdjustedAmounts[Field.INPUT].raw.toString())
```

Comment:
I think we should extract this transaction construction logic into a unit testable thing since we definitely need to verify the arguments in the tx we construct (todo)

Reply:
want to get this merged, so going to resolve, but i agree